### PR TITLE
Laura | JS_adagrams

### PR DESCRIPTION
**Pull request by lauracodecreations**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | The 'if statements' work the same, the 'while' loop, and 'push' and 'pop' methods for arrays. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | I made sure to specify Javascript and use JS terminology like objects instead of hashes. 
What was a challenge you were able to overcome on this assignment? | solving the ties was challenging because the tools I used in Ruby where not available in Javascript or even a tool with similar function. For example, for an object that has a k-v pair, return the key that has a specific number in its value. As an illustration, getting all the keys that have a value of 2 in this hash: { X: 32, XXXXXX: 16, XXXX: 2, XX: 2 }. I overcame this problem, by using the 'filter' function which was the only function that I found that returns an array of multiple elements if the multiple items meet the requirements. I used the length of this outputted array to determine if I had a tie in my list of words. 
What is your favorite thing about learning a new programming language? | My favorite thing about learning a new language was thinking about new ways of solving the problem when the tools you are used to do not exist in JS.  
What is your least favorite thing about learning a new programming language? | Getting used to all the new terminology like callback functions and arrow functions. 
Do you have any recommendations on how we could improve this project for the next cohort? | I liked how we went to an old project because it allowed me to optimize it even more with the new skills I have developed.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,144 @@
...
+      let freqInHand = letterCountsInHand[letter]
+      if (freqInHand == undefined) {
+        answer = false;
+      } else if (freqInput > freqInHand) {
+        answer = false;
+      }
+    });
+    if (answer == false) {
+      return answer
+    } else {
```
Comment (Hamled) [inline src/adagrams.js:77]: This final if statement could be written as `return answer;` if the declaration on line 65 is changed to `let answer = true;`.

**Comment by Hamled** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | :white_check_mark:
Small commits with meaningful commit messages | :white_check_mark:
**Code Requirements** |
`drawLetters` method | :white_check_mark:<br>This is a really interesting way to pick 10 random letters without having to copy and modify the original array of letters. I like it!
Uses appropriate data structure to store the letter distribution | :white_check_mark:
All tests for `drawLetters` pass | :white_check_mark:
`usesAvailableLetters` method | :white_check_mark:
All tests for `usesAvailableLetters` pass | :white_check_mark:
`scoreWord` method | :white_check_mark:
Uses appropriate data structure to store the letter scores | Mostly.<br>I think this code could also have been written with an object (used like a Ruby hash) for storing the letter scores. However, it's definitely true that JavaScript doesn't make the syntax for that approach nearly as nice as the Ruby version.
All tests for `scoreWord` pass | :white_check_mark:
`highestScoreFrom` method | :white_check_mark:
Appropriately handles edge cases for tie-breaking logic | :white_check_mark:
All tests for `highestScoreFrom` pass | :white_check_mark:
**Overall** | Looks good!
